proposals/tx-size.md

```diff
@@ -0,0 +1,102 @@
+# Transaction Size Calculation Methods
+
+## Summary
+
+This proposal introduces two new getter methods to the Hedera SDKs (JavaScript, Java, Go, etc.) to allow **calculating transaction sizes before submission**. These methods are essential for:
+
+- Determining transaction fees more accurately before execution
+- Validating transactions against network size limits
+- Supporting advanced transaction batching scenarios
+
+The proposed methods are:
+
+- `size`: Returns the total transaction size in bytes after protobuf encoding
+- `bodySize`: Returns just the transaction body size in bytes after encoding
+
+## New API
+
+### `Transaction.size`
+
+**Type**: `Promise<number>`  
+**Description**: Returns the total size in bytes of the transaction after protobuf encoding. This includes signatures, transaction body, and all other transaction components.
+
+Since the actual wire format can only be known after encoding, this method returns a Promise.
+
+**Implementation**:
+
+```javascript
+get size() {
+    this._requireFrozen();
+    return this._makeRequestAsync().then(
+        (request) =>
+            HieroProto.proto.Transaction.encode(request).finish().length,
+    );
+}
+```
+
+### `Transaction.bodySize`
+
+**Type**: `number`  
+**Description**: Returns the size in bytes of just the transaction body after protobuf encoding. This is useful for understanding the core transaction size without signatures. Implementation uses Node Account Id `0.0.0` as the node account id doesn't matter for the transaction body size.
+
+**Implementation**:
+
+```javascript
+get bodySize() {
+    const body = this._makeTransactionBody(AccountId.fromString("0.0.0"));
+    return HieroProto.proto.TransactionBody.encode(body).finish().length;
+}
+```
```

Comment:
I'm a bit confused on why `bodySize` returns just a `number` while `size` returns a `Promise<number>`? Would it make sense to add a synchronous method to get the full request body?

Reply:
Since the getter for the body calls `_makeRequestAsync` in the body and it is an asynchronous function, the entire function must return a Promise.